test/helpers.h

```diff
@@ -92,4 +92,18 @@ class PreprocessorHelper
     static std::string getcode(Preprocessor &preprocessor, const std::string &filedata, const std::string &cfg, const std::string &filename, Suppressions *inlineSuppression = nullptr);
 };
 
+/* designated initialization helper
+    Usage:
+    struct S
+    {
+        int i;
+    };
+
+    const auto s = dinit(S,
+        $.i = 1
+    );
+ */
+#define dinit(T, ...) \
+    ([&]{ T ${}; __VA_ARGS__; return $; }())
```

Comment:
So doing `dinit(S, $.i = 1, $.j = 2)` wont evaluate the initialization in order.

Reply:
That should not be an issue in most of the cases. There are a few where that could be a problem though.